lib/matplotlib/axes/_axes.py

```diff
@@ -6871,6 +6880,15 @@ def hist2d(self, x, y, bins=10, range=None, normed=False, weights=None,
           `.colors.PowerNorm`.
         """
 
+        if density is not None and normed is not None:
+            raise ValueError("kwargs 'density' and 'normed' cannot be used "
+                             "simultaneously. Please only use 'density', "
+                             "since 'normed' is deprecated.")
+        if normed is not None:
+            cbook.warn_deprecated("3.1", name="'normed'", obj_type="kwarg",
+                                  alternative="'density")
+
+        normed = bool(density) or bool(normed)
         h, xedges, yedges = np.histogram2d(x, y, bins=bins, range=range,
```

Comment:
I don't think we should call the argument `normed` if we are going to change to `density` later.  

This could also use a comment in here about how we will need to switch to `density` when np1.15 becomes our minimal version.  Otherwise if I was looking at this code I'd have no idea why you are deprecating the name `normed` when that's what numpy is using.  Indeed reviewing this I had to see @anntzer 's comment: https://github.com/matplotlib/matplotlib/issues/11070#issuecomment-422411546 to understand why you were doing this.  Should probably be in the PR description as well to save folks back referencing.

Reply:
Thanks for the comments. I was trying to be consistent with [`hist`](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.hist.html). #11070 was opened to have API consistency. I can also change `hist` to be consistent with your suggestions.

Reply:
I'm fine with the change proposed here , but only because numpy is moving to using `density` as well (in np1.15).  So the PR should be clear why we are doing this.

Reply:
So you still have `normed = bool(density) or bool(normed)`, I think you should be populating and using `density` *not* `normed`.  

I think there should be a comment in the *code* here for future reference to explain this.

Reply:
I misunderstood about your comment on `density`. Let me fix it right away.